detox/__tests__/setupJest.js

```diff
@@ -1,94 +1,3 @@
 jest.mock('proper-lockfile');
 jest.mock('signal-exit');
 jest.mock('../src/logger/DetoxLogger');
-
-const path = require('path');
-
-const _ = require('lodash');
-const yargs = require('yargs');
-
-function callCli(modulePath, cmd) {
-  return new Promise((resolve, reject) => {
-    const originalModule = require(path.join(__dirname, '../local-cli', modulePath));
-    const originalHandler = originalModule.handler;
-    const spiedModule = {
-      ...originalModule,
-      handler: async program => {
-        try {
-          return await originalHandler(program);
-        } catch (e) {
-          reject(e);
-        } finally {
-          resolve();
-        }
-      }
-    };
-
-    return yargs
-      .scriptName('detox')
-      .parserConfiguration({
-        'boolean-negation': false,
-        'camel-case-expansion': false,
-        'dot-notation': false,
-        'duplicate-arguments-array': false,
-      })
-      .command(spiedModule)
-      .exitProcess(false)
-      .fail((msg, err) => reject(err || msg))
-      .parse(cmd, (err) => err && reject(err));
-  });
-}
-
-/**
- * A syntactic-sugar wrapper for performing expectations while a tested async method/func is
- * suspended, based on a "suspending mock".
- *
- * It applies the following technique:
- *
- * 1. Setting up an (async) mock function -- which the method under test normally calls, such that it would suspended
- *    until a provided inner callback is completed.
- * 2. Calling the method under test, making expectations inside a callback using a custom `callSuspended` function.
- *
- * @example
- * await withSuspendingMock(innerMockObj, 'mockedMethodName', async ({ callSuspended }) => {
- *   await callSuspended(uut.methodUnderTest(), async () => {
- *     expect(uut.nextSolarEclipse()).toEqual( Date.parse('20 Apr 2023') );
- *
- *     return 'result-of-suspended-method'; // (Optional)
- *   }); // Resolve suspended method
- * }); // Await methodUnderTest's promise
- */
-async function withSuspendingMock(obj, func, callback) {
```

Comment:
🙏  I'm always sorry to remove such well-worked code parts, but there's little sense in reinventing the wheel.

What you tried to achieve was trivial enough if you would have re-formulated it in Jest DSL:

```js
it('should change intermediate status to "cleanup"', async () => {
  // making sure that the test is not marked as PASSED without running the assertion below
  expect.assertions(1);
  await facade.init();

  ipcClient().dispose.mockImplementation(async () => {
    // running an expectation while `facade.cleanup()` is running
    expect(facade.getStatus()).toBe('cleanup');
  });

  await facade.cleanup(); // launching the sequence
});
```

Reply:
nice! don't worry about it! makes perfect sense

Reply:
Oh BTW, be sure to also `expect()` that `ipcClient().dispose` has indeed been called......

Reply:
Oh that's `expect.assertions(1)`... got it